### PR TITLE
ES-1850: Refactor virtual-node plugin onto Corda SDK

### PR DESCRIPTION
**Pull request by anton-subbotin**
### [ES-1850](https://r3-cev.atlassian.net/browse/ES-1850)

Move the logic of `virtual-node platform-migration` CLI command to Corda SDK. The intention of this refactor is to leave the existing behaviour, therefore no implementation changes were made.

- Upgrade tests, where the command is used: https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda5-upgrade-tests/detail/feature%2Fasubbotin%2FES-1850%2Fuse-corda-os-docker-registry/10/pipeline - fails due to known issue; the stages with virtual node DB migration passed
- Plugins smoke tests: https://ci02.dev.r3.com/job/Corda5/job/corda-runtime-os-combined-worker-plugins-smoke-tests/job/PR-6068/3/

[ES-1850]: https://r3-cev.atlassian.net/browse/ES-1850?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tools/plugins/virtual-node/src/main/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigration.kt [hunk 1]
```diff
@@ -102,33 +81,14 @@ class PlatformMigration(
...
-                LiquibaseFileAndSchema("net/corda/db/schema/vnode-uniqueness/db.changelog-master.xml", "vnode_uniq_"),
-                LiquibaseFileAndSchema("net/corda/db/schema/vnode-vault/db.changelog-master.xml", "vnode_vault_")
-            ).forEach { fileAndSchema ->
-                holdingIdsToMigrate.forEach { holdingId ->
-                    generateSql(fileWriter, holdingId, fileAndSchema)
-                }
-            }
-        }
-    }
+    override fun call(): Int {
```
Comment (tlawson3) [inline tools/plugins/virtual-node/src/main/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigration.kt:59]: why has this changed from run to call?

File: tools/plugins/virtual-node/src/main/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigration.kt [hunk 2]
```diff
@@ -102,33 +81,14 @@ class PlatformMigration(
...
-            }
-
-            connection.use {
-                val lb = config.liquibaseFactory(fileAndSchema.filename, database)
-                liquibaseSchemaUpdater.update(lb, fileWriter)
-            }
+            generator.generateVNodeMigrationSqlFile(holdingIdFilename, outputFilename, jdbcConnectionParams)
         }
+
+        return ExitCode.OK
```
Comment (tlawson3) [inline tools/plugins/virtual-node/src/main/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigration.kt:67]: Is this related to the run/call change, does call require a return value?

File: tools/plugins/virtual-node/src/pluginSmokeTest/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigrationSmokeTest.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                "-u=$postgresUser",
+                "-p=$postgresPassword",
+                "-i=${holdingIdsFile.absolutePath}",
+                "-o=${outputFile.absolutePath}"
+            )
+        assertEquals(0, exitCode)
+        val updateDatabaseScriptLines = outputFile.readLines().filter {
+            it.contains("-- Update Database Script")
+        }
+        assertEquals(3 * validHoldingIds.size, updateDatabaseScriptLines.size)
```
Comment (tlawson3) [inline tools/plugins/virtual-node/src/pluginSmokeTest/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigrationSmokeTest.kt]: assertThat seems to be the preferred approach in runtimeos

File: tools/plugins/virtual-node/src/main/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigration.kt (hunk 1 above)
Comment (anton-subbotin) [reply tools/plugins/virtual-node/src/main/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigration.kt:59]: Chose it to return an exit code and check it in tests

File: tools/plugins/virtual-node/src/pluginSmokeTest/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigrationSmokeTest.kt (hunk 3 above)
Comment (anton-subbotin) [reply tools/plugins/virtual-node/src/pluginSmokeTest/kotlin/net/corda/cli/plugins/vnode/commands/PlatformMigrationSmokeTest.kt]: done

**Review by tlawson3 — APPROVED** [review]

**Comment by anton-subbotin** [comment]
build e2e

**Review by vkolomeyko — APPROVED** [review]
LGTM (from @corda/rest PoV)

**Review by seanbrereton — APPROVED** [review]
LGTM - Infra team

**Review by wzur-r3 — APPROVED** [review]
LGTM for Jenkins pipeline change

**Comment by anton-subbotin** [comment]
build e2e

**Pull request merged**